src/main/kotlin/com/progreizh/animaliste/services/AnimalService.kt

```diff
@@ -0,0 +1,21 @@
+package com.progreizh.animaliste.services
+
+import com.progreizh.animaliste.repositories.AnimalRepository
+import org.springframework.stereotype.Service
+
+@Service
+class AnimalService(private val animalRepository: AnimalRepository) {
+    /**
+     * Récupère les éspèces différentes des animaux.
+     */
+    fun getSpecies() : Set<String>{
+        val animals = animalRepository.findAll()
+        val species = HashSet<String>()
+        animals.forEach {
+            if(!species.contains(it.specie)){
+                species.add(it.specie);
+            }
+        }
+        return species;
+    }
```

Comment:
Plutôt question que suggestion. Pourquoi ne pas utiliser directement `AnimalRepository`, peut être en ajoutant une méthode.

Cf. : 
- https://stackoverflow.com/questions/32108953/how-to-return-only-specific-fields-for-a-query-in-spring-data-mongodb
- https://docs.mongodb.com/compass/master/query/project

Reply:
Je récupère des strings et non des animaux